btcjson/chainsvrresults.go

```diff
@@ -398,6 +399,63 @@ type GetRawMempoolVerboseResult struct {
 	Depends          []string `json:"depends"`
 }
 
+// GetZmqNotificationResult models the data returned from the getzmqnotifications command.
+type GetZmqNotificationResult []struct {
+	Type          string   // Type of notification
+	Address       *url.URL // Address of the publisher
+	HighWaterMark int      // Outbound message high water mark
+}
+
+func (z *GetZmqNotificationResult) MarshalJSON() ([]byte, error) {
+	var out []map[string]interface{}
+	for _, notif := range *z {
+		out = append(out,
+			map[string]interface{}{
+				"type":    notif.Type,
+				"address": notif.Address.String(),
+				"hwm":     notif.HighWaterMark,
+			})
+	}
+	return json.Marshal(out)
+}
+
+// UnmarshalJSON satisfies the json.Unmarshaller interface
+func (z *GetZmqNotificationResult) UnmarshalJSON(bytes []byte) error {
+	type basicNotification struct {
+		Type    string
+		Address string
+		Hwm     int
+	}
+
+	var basics []basicNotification
+	if err := json.Unmarshal(bytes, &basics); err != nil {
+		return err
+	}
+
+	var notifications GetZmqNotificationResult
+	for _, basic := range basics {
+
+		address, err := url.Parse(basic.Address)
+		if err != nil {
+			return err
+		}
+
+		notifications = append(notifications, struct {
+			Type          string
+			Address       *url.URL
+			HighWaterMark int
```

Comment:
Similar to a [comment](https://github.com/btcsuite/btcd/pull/1555#discussion_r429568250) I made on your previous PR:

Can we do something about reusing the struct definition of `GetZmqNotificationResult`? If a new field is added to the response, we'll otherwise need to make the change in several places.

Reply:
Following up again on this. I recently found a way to address this, and have been using the technique in some of my RPC PRs. [Here](https://github.com/btcsuite/btcd/pull/1641/files#diff-1a3cd13f1a92a486f6158742c1759b8eR453-R498) is an example. What do you think?